### DISCUSSION TITLE
What is the difference of Coherence Community Edition between major version 23 and 14

### DISCUSSION THREAD

**Discussion by takeaki-m** [oracle/coherence#116] [Q&A] [answered]
Hi. I am considering using the community edition of Oracle Coherence for development. 

it appears that there are two major versions of the Community Edition, 14 and 23. Is there any difference between them? 
Are there any conditions of use, such as in certain situations it is better to use one of the versions?
if there were documents that shows the difference, I would greatly appreciate you when you share them.

I have checked some documents like below, but I can't tell the difference.
https://docs.oracle.com/en/middleware/standalone/coherence/14.1.1.2206/develop-applications/index.html#Oracle%C2%AE-Coherence
https://coherence.community/latest/23.09/docs/#/docs/about/01_overview
https://coherence.community/latest/14.1.1.0/docs/#/docs/about/01_overview

The only one difference I checked is the prerequisite

- 23
> Java - JDK 17 or higher
> Maven - 3.8.5 or higher

- 14
> Java - jdk8 or higher
> Maven - 3.6.3 or higher

https://coherence.community/latest/23.09/docs/#/docs/about/03_quickstart
https://coherence.community/latest/14.1.1.0/docs/#/docs/about/03_quickstart

**Comment by thegridman** [comment, marked as answer]
@takeaki-m There are actually three versions

- 23.09 the latest version (until 24.03 in a few weeks)
- 22.06 the latest LTS version
- 14.1.1-0-16 the old Java 8 compatible 14.x release

Personally unless you are stuck on Java 8 I would ignore the 14.1.1 version, it is old has has none of the newer features from the later releases.

Coherence CE non-LTS releases are supported for six months, so 23.09 is supported until 24.03 comes out. The LTS releases, such as 22.06 are supported for longer, and hence receive patched, CVE fixes etc.

23.09 (and soon to be 24.03) use the JEE jakarta package names so if you are using Coherence with something like the latest versions of Helidon, Spring, Micronaut or other frameworks that also use Jakarta EE package names you will need to use this version.

22.06 uses JEE javax package names, so works with older versions of frameworks.

As you have already seen 
- 22.06 requires Java 11 (but work and is tested on later versions).
- 23.09 and 24.03 requires Java 17 (but work and is tested on later versions) Some modules in 24.03 will require Java 21, such as the Helidon 4 gRPC server module

**Comment by aseovic** [comment]
@takeaki-m Basically, 14.1.1.x version is the first open source release, and tracks the commercial 14.1.1.x release. It requires Java 8, as you noted.

We adopted YY.MM versioning schema (a la Ubuntu) for all subsequent CE releases, with releases every 6 months. Initially we started with releases in June and December, such as 21.12 and 22.06, but have since switched to March/September release schedule, with the latest release being 23.09, and with 24.03 just around the corner, in a few weeks 😉

So which one should you use? It depends…

In general, the latest, because that’s where all the patches go, but there are multiple “latest” releases. There is absolute latest, 23.09, which will soon be obsolete and replaced by 24.03, but neither of these are LTS releases and will only be supported until the next release comes out, so you may need to upgrade frequently. If you want latest and greatest, and don’t mind being on the bleeding edge, that’s what you should use.

The latest LTS (Long-term supported) release is 22.06, so if you need stability, that’s probably your best bet. The next LTS release will be 24.09 in September, which will be likely identical to the upcoming 24.03 release, with some stabilization and hardening work between now and September. If that works for you, you should probably start with 23.09/24.03, and upgrade to an LTS 24.09 once it’s released.

However, there is another consideration: which version of Java, Jakarta and application frameworks (if any) you plan to use.

If you plan to use the latest version of Spring, Helidon or Micronaut, they all depend on Jakarta EE 9+, and Java 17+. In that case, 23.09/24.03 is your only option.

On the other hand, if you need to use Java 11 and/or Jakarta EE 8, you need to use 22.06.

Unfortunately, switch from Jakarta EE 8 to 9 required package name changes, so it was impossible to create a release that works with both.

Hope this answers your question, but you should also review release notes for each release to better understand both the requirements they have, and the features and improvements they provide.

**Reply by aseovic** [reply]
Oh — the only reason you should use 14.1.1.x is if you need to use Java 8. In any other case you are better off using one of the more recent releases.

**Comment by takeaki-m** [comment]
@thegridman @aseovic Thank you for your quick and comprehensive answers!! I understand the maintenance policies, coherence versioning, and dependencies with Java and Jakarta for each version. 
So, 22.06 is my first choice for now😊

There's one more thing I'd like to ask you about 14.1.1.x.

Is 14.1.1.x still under LTS? Or is it no longer receiving updates for security vulnerabilities?

I assume that 14.1.1.x is no longer updated with new features, but I'm not sure if it still receives security support.

**Comment by thegridman** [comment]
@takeaki-m 14.1.1 is still LTS. You are right, it is not getting new features, but still gets bug fixes and CVE fixes

**Comment by takeaki-m** [comment]
@thegridman I'm grateful for your reply.  I got that 14.1.1 is still under maintaince.

**Comment by takeaki-m** [comment]
Thank you both for your quick answers. 
I really want to mark both of your answers as ANSWER, but I can only mark on due to GitHub restriction, so I'll mark the one ahead of time. 

Thank you also for your daily code maintenance!!
